tests/fast-integration/utils/index.js

```diff
@@ -692,7 +715,86 @@ function getEnvOrThrow(key) {
   return process.env[key];
 }
 
+function wait(fn, checkFn, timeout, interval) {
+  return new Promise((resolve, reject) => {
+      const th = setTimeout(function() {
+        debug("wait timeout");
+        done(reject, new Error("wait timeout"));
+      }, timeout);
+      const ih = setInterval(async function() {
+          let res;
+          try { res = await fn(); } 
+          catch(ex) { res = ex; }
+          checkFn(res) && done(resolve, res);
+      }, interval);
+      
+      function done(fn, arg) {
+          clearTimeout(th);
+          clearInterval(ih);
+          fn(arg);
+      }
+  });
+}
+
+async function ensureApplicationMapping(name, ns) {
```

Comment:
I would rather keep it in commerce-mock fixture. Utils are a more generic functions. The same with patching application gateway.

Reply:
Requires further refactoring of utils and I think we should extract from utils all k8s related functions to separate client initialised from kubeconfig.